### ISSUE TITLE
NET8NativeContainers - I'm missing something?

### ISSUE THREAD

**Issue by chdev77** [aws-samples/serverless-dotnet-demo#40]
Great work on your YT and repo, much appreciated.

Trying to get Net8NativeContainers to work.
**Steps**
Clone repo
Build NET8NativeContainers.sln
cmd: docker build -f src/NET8NativeContainers/PutProduct/Dockerfile -t serverless-sample .

Any ideas?

![image](https://github.com/aws-samples/serverless-dotnet-demo/assets/7691322/81b1b08c-c8f5-44a1-bca4-d35befa00eac)

**Comment by Beau-Gosse-dev** [comment]
Hi @chdev77 I also tried to build this image on my Windows machine but it seems to hang. I would say that native AOT and ARM64 (as seen in the screenshot) may not work well on Windows x86 machine. What OS and CPU architecture are you using?

**Comment by chdev77** [comment]
Windows 10 x86

**Comment by ppittle** [comment]
```
qemu: uncaught target signal 11 (Segmentation fault) - core dumped
```
Was able to reproduce this on my Win 10 in powershell and wsl.
```
Docker Architecture: x86_64
Docker version 24.0.7
```

Looking at the .NET Documentation, looks like it's a virtualization error: https://github.com/dotnet/core/blob/main/release-notes/8.0/supported-os.md#qemu

> .NET 8 is not supported being run (emulated) via [QEMU](https://www.qemu.org/). QEMU is used, for example, to emulate Arm64 containers on x64, and vice versa.

@Beau-Gosse-dev @jeastham1993 - are these docker images intended to be built on a Arm64 (ie Graviton) machine or can they be built on an Amd64 machine?

Do the docker files need to be updated as advised here: https://devblogs.microsoft.com/dotnet/improving-multiplatform-container-support/

**Comment by Beau-Gosse-dev** [comment]
Yes, we only intended to support same-architecture builds. Adding support as outlined in that blog would be a great addition if we can get it to work for native AOT Lambdas.
